src/main/java/be/seeseemelk/mockbukkit/MockPlugin.java

```diff
@@ -33,4 +35,101 @@ protected MockPlugin(@NotNull JavaPluginLoader loader, @NotNull PluginDescriptio
 		super(loader, description, dataFolder, file);
 	}
 
+	public static Builder builder()
+	{
+		return new Builder();
+	}
+
+	public static class Builder
+	{
+
+		private Optional<Runnable> onLoad = Optional.empty();
+		private Optional<Runnable> onEnable = Optional.empty();
+		private Optional<Runnable> onDisable = Optional.empty();
+		private String pluginName = "MockPlugin";
+		private String pluginVersion = "1.0.0";
+
+		private Builder()
+		{
+		}
+
+		public Builder withOnLoad(Runnable onLoad)
+		{
+			this.onLoad = Optional.of(onLoad);
+			return this;
+		}
+
+		public Builder withOnEnable(Runnable onEnable)
+		{
+			this.onEnable = Optional.of(onEnable);
```

Comment:
It seems that we also support null here, if so we should use Optional.ofNullable here, otherwise a NPE is thrown.